packages/amplify_core/lib/src/types/api/rest/http_payload.dart

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:async/async.dart';
+
+/// {@template amplify_core.http_payload}
+/// An HTTP request's payload.
+/// {@endtemplate}
+class HttpPayload extends StreamView<List<int>> {
+  /// {@macro amplify_core.http_payload}
+  factory HttpPayload([Object? body]) {
+    if (body == null) {
+      return const HttpPayload.empty();
+    }
+    if (body is String) {
+      return HttpPayload.string(body);
+    }
+    if (body is List<int>) {
+      return HttpPayload.bytes(body);
+    }
+    if (body is Stream<List<int>>) {
+      return HttpPayload.streaming(body);
+    }
+    if (body is Map<String, String>) {
+      return HttpPayload.fields(body);
+    }
+    throw ArgumentError('Invalid HTTP payload type: ${body.runtimeType}');
+  }
+
+  /// An empty HTTP body.
+  const HttpPayload.empty() : super(const Stream.empty());
+
+  /// A UTF-8 encoded HTTP body.
+  HttpPayload.string(String body, {Encoding encoding = utf8})
+      : super(LazyStream(() => Stream.value(encoding.encode(body))));
+
+  /// A byte buffer HTTP body.
+  HttpPayload.bytes(List<int> body) : super(Stream.value(body));
+
+  /// Form-encodes the body.
+  HttpPayload.fields(Map<String, String> body, {Encoding encoding = utf8})
```

Comment:
I'd prefer `form` or `formValues` unless there's some precedence for this name

Reply:
I'll go with `formValues` in next rev. Wasn't wed to the name, went with that bc called bodyFields in http https://pub.dev/documentation/http/latest/http/Request-class.html and I thought "fields" was the best way to mirror that, but... not the best name by itself.